File: 0948-sort-an-array/0948-sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        int left =0, right =nums.length-1;
        sort(nums,left,right);
        return nums;
    }
    public void sort(int[] nums,int left, int right){
        if(left < right){
            int mid = (right+left)/2;
            sort(nums,left,mid);
            sort(nums,mid+1,right);
            merge(nums,left,mid,right);
        }
    }
    public void merge(int[] nums,int left,int mid,int right){
        int n1 = mid-left+1;
        int n2 = right - mid;
        int L[] = new int[n1];
        int R[] = new int[n2];

        for(int i=0;i<n1;i++){
            L[i] = nums[left+i];
        }
        for(int i=0;i<n2;i++){
            R[i] = nums[mid+1+i];
        }
        int i=0,j=0;
        int k= left;
        while( i<n1 && j<n2 ){
            if(L[i] < R[j]){
                nums[k] = L[i];
                i++;
            }else{
                nums[k] = R[j];
                j++;
            }
            k++;
        }
        while(i<n1){
                nums[k] = L[i];
                i++;
                k++;
         }
         while(j<n2){
                nums[k] = R[j];
                j++;
                k++;
         }
    }
}

File: 0948-sort-an-array/0948-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        return self.merge_sort(nums)
    def merge_sort(self,cities):

        if len(cities) <= 1:
            return cities

        # Divide the list into two halves
        mid = len(cities) // 2
        left = cities[:mid]
        right = cities[mid:]

        # Recursively apply merge_sort to each half
        left = self.merge_sort(left)
        right = self.merge_sort(right)

        # Merge the two sorted halves
        result = []
        i = j = 0

        while i < len(left) and j < len(right):
            if left[i] <= right[j]:
                result.append(left[i])
                i += 1
            else:
                result.append(right[j])
                j += 1

        # Append remaining elements (if any)
        result.extend(left[i:])
        result.extend(right[j:])
        
        return result
        